csrc/multidevice/collective.h

```diff
@@ -0,0 +1,168 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#pragma once
+#ifdef USE_DISTRIBUTED
+
+#include <multidevice/communicator.h>
+#include <multidevice/multidevice.h>
+
+namespace nvfuser {
+
+/*
+  This struct gathers all the parameters necessary for the
+  construction a collective
+*/
+struct TORCH_CUDA_CU_API CollectiveParams {
```

Comment:
I wonder if this should be renamed just like `CommParams`. There's nothing specific to collective communication patterns (except for `root`, which may not be used for collective ops), and indeed we subclass this to represent a point-to-point communication as well.

Reply:
I renamed it as you suggest